vars/MavenCommand.groovy

```diff
@@ -0,0 +1,142 @@
+def class MavenCommand {
+
+    def steps
+
+    boolean returnStdout = false
+
+    String settingsXmlPath = ''
+    List mavenOptions = []
+    Map properties = [:]
+    String logFileName = null
+    List profiles = []
+
+    MavenCommand(steps){
+        this.steps = steps
+    }
+
+    MavenCommand(steps, List defaultOpts){
+        this(steps)
+        this.mavenOptions.addAll(defaultOpts)
+    }
+
+    def run(String goals) {
+        def cmd = "mvn -B"
+        if(this.settingsXmlPath) {
+            cmd += " -s ${this.settingsXmlPath}"
+        }
+        if(this.mavenOptions.size() > 0){
+            cmd += " ${this.mavenOptions.join(' ')}"
+        }
+        cmd += " ${goals}"
+        if(this.profiles.size() > 0){
+            cmd += " -P${this.profiles.join(',')}"
+        }
+        if(this.properties.size()){
+            cmd += " ${this.properties.collect{ it.value != '' ? "-D${it.key}=${it.value}" : "-D${it.key}" }.join(' ')}"
+        }
+        if(this.logFileName){
+            cmd += " | tee \$WORKSPACE/${logFileName} ; test \${PIPESTATUS[0]} -eq 0"
+        }
+
+        if(this.returnStdout){
+            return steps.sh(script: cmd, returnStdout: this.returnStdout)
+        }
+        steps.sh cmd
+    }
+
+    def runClean() {
+        return run('clean')
+    }
+
+    def runPackage(boolean clean = false) {
+        return run(clean ? 'clean package' : 'package')
+    }
+    
+    def runInstall(boolean clean = false) {
+        return run(clean ? 'clean install' : 'install')
+    }
+
+    def runVerify(boolean clean = false) {
+        return run(clean ? 'clean verify' : 'verify')
+    }
+
+    def runDeploy(boolean clean = false, String deployRepository = ''){
+        if(deployRepository){
+            withProperty('altDeploymentRepository', "runtimes-artifacts::default::${deployRepository}")
+            withProperty('enforcer.skip', true)
+        }
+        return run(clean ? 'clean deploy' : 'deploy')
+    }
+
+    def runDeployLocally(String localDeployFolder, boolean clean = false){
+        if(localDeployFolder){
+            withProperty('altDeploymentRepository', "local::default::file://${localDeployFolder}")
+        }
+        return run(clean ? 'clean deploy' : 'deploy')
+    }
```

Comment:
How much is this "syntactic sugar" needed? Personally, I would be fine with just `def run(String goals)` and maybe `withDeployRepository` and `withLocalDeployFolder` to compensate for the `runDeploy()` methods.

But I don't have really strong opinion on this; your call.

Reply:
right, maybe not that useful. gonna remove this and add the with* methods instead